test/extensions/filters/network/thrift_proxy/integration_test.cc

```diff
@@ -119,6 +119,42 @@ class ThriftConnManagerIntegrationTest
       }
     });
 
+    std::tie(std::ignore, std::ignore, std::ignore, payload_passthrough_) = GetParam();
+
+    if (payload_passthrough_) {
+      config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+        ASSERT_TRUE(bootstrap.mutable_static_resources()->listeners_size());
+        auto* listener = bootstrap.mutable_static_resources()->mutable_listeners(0);
+        ASSERT_TRUE(listener);
+        ASSERT_TRUE(listener->filter_chains_size());
+
+        envoy::config::listener::v3::Filter* filter = nullptr;
+        auto* filter_chain = listener->mutable_filter_chains(0);
+        for (ssize_t i = 0; i < filter_chain->filters_size(); i++) {
+          if (filter_chain->mutable_filters(i)->name() == "thrift") {
```

Comment:
This seems a very round-about way to augment the config; can't you set this when defining the thrift config to begin with? What about utilities in https://github.com/envoyproxy/envoy/blob/master/test/config/utility.h? Maybe add a `getFilterByName()`?

Reply:
> can't you set this when defining the thrift config to begin with? 

Since the thrift config is defined in  `static void SetUpTestSuite()`, i cannot set it according to the test param.

>What about utilities in https://github.com/envoyproxy/envoy/blob/master/test/config/utility.h? Maybe add a getFilterByName()?

I'm considering to add `void addConfigModifier(ThriftModifierFunction function)`,  alongside the `void addConfigModifier(HttpModifierFunction function)`.

Reply:
I think it could be made a bit more generic and not special case on "thrift"; it could just scan for a given network filter config and apply some transformer on it.